common/include/pcl/register_point_struct.h

```diff
@@ -265,7 +265,7 @@ namespace pcl
       inline const name operator+ (const name& lhs, const name& rhs)   \
       { name result = lhs; result += rhs; return (result); }           \
       inline const name operator+ (const float& scalar, const name& p) \
-      { name result = p; result += scalar; return (result); }          \
+      { float result = scalar; result +=p; return (result); }          \
```

Comment:
The return type of this function is `name` (i.e. the point type), however with your change we return `float`. I'm not sure I understand how this is supposed to work.

Reply:
@Swati-007 

In the case of #2465, `name` is `pcl::PointXY`.

For the minus operator, the following code segments are relevant (a good starting point):
https://github.com/PointCloudLibrary/pcl/blob/9467b944ee3e30bb4cf8445d861c280f0fba1020/common/include/pcl/register_point_struct.h#L155-L163

https://github.com/PointCloudLibrary/pcl/blob/9467b944ee3e30bb4cf8445d861c280f0fba1020/common/include/pcl/register_point_struct.h#L218-L220

https://github.com/PointCloudLibrary/pcl/blob/9467b944ee3e30bb4cf8445d861c280f0fba1020/common/include/pcl/register_point_struct.h#L285-L286